docs/content/development/extensions-core/sample-client-sslcontext.md

```diff
@@ -0,0 +1,27 @@
+---
+layout: doc_page
+---
+
+## Sample SSLContext Provider Module
```

Comment:
Yes it is necessary to use an extension for internal communication over TLS. I have mentioned his in the tls-support.md page under "Druid's internal communication over TLS" header.
For other comments - https://github.com/druid-io/druid/pull/4270#discussion_r120215855